### PR TITLE
Change: Reorganize policy

### PR DESCRIPTION
**Pull request by nickanderson**
Not all policy that ships with CFEngine is applicable or even useful to
community users. This work is to seperate the enterprise specific policy from
the core policy and only load the enterprise specific policy when necessary.

Additionally this is intended to make user entry points more clear, and easier to upgrade the policy framework in the future

### CODE REVIEW

**Comment by nickanderson** [comment]
trigger build

File: cfe_internal/CFE_cfengine.cf [hunk 1]
```diff
@@ -23,176 +23,13 @@ bundle agent cfe_internal_management
...
-      "rotate_outputs"         usebundle => prunedir("@(def.cfe_log_dirs)", "30"),
-                               comment   => "Delete outputs/* files older than 30 days";
+      "any"
+        usebundle => "cfe_internal_core_main",
+        comment => "Activate policies related to basic cfengine operations";
+
+    enterprise_edition::
+      "any"
+        usebundle => "cfe_internal_enterprise_main",
+        comment => "Activate policies related to CFEngien Enterprise operations";
```
Comment (tzz) [inline cfe_internal/CFE_cfengine.cf]: typoe

File: cfe_internal/CFE_cfengine.cf [hunk 2]
```diff
@@ -23,176 +23,13 @@ bundle agent cfe_internal_management
...
-      # CFEngine generates internal log files that need to be rotated.
-      # Have a look at def.cf to enable rotation of these files
-      "rotate_promise_summary" usebundle => logrotate("@(def.cfe_log_files)", "1m", "10"),
-                               comment   => "Rotate log files once their size reach 1MB, keep 10 versions";
-
-      "rotate_outputs"         usebundle => prunedir("@(def.cfe_log_dirs)", "30"),
-                               comment   => "Delete outputs/* files older than 30 days";
+      "any"
+        usebundle => "cfe_internal_core_main",
+        comment => "Activate policies related to basic cfengine operations";
```
Comment (tzz) [inline cfe_internal/CFE_cfengine.cf]: cAPITALIZE

File: cfe_internal/core/limit_robot_agents.cf [hunk 3]
```diff
@@ -0,0 +1,77 @@
+##################################################################
+#
+# cfe_internal_limit_rebot_agents
```
Comment (tzz) [inline cfe_internal/core/limit_robot_agents.cf]: typoe

File: cfe_internal/core/limit_robot_agents.cf [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+#  - kill CFE processes and restart it when the process grown
```
Comment (tzz) [inline cfe_internal/core/limit_robot_agents.cf]: "grown" typoe

File: cfe_internal/core/log_rotation.cf [hunk 5]
```diff
@@ -0,0 +1,24 @@
+##################################################################
+#
+# cfe_internal_log_rotation
+#  - Rotate logs and clean up old files so the disk doesnt get full
+#
+##################################################################
+
+
+bundle agent cfe_internal_log_rotation
+# @brief Manage CFEngine log files so they dont fill up disks
```
Comment (tzz) [inline cfe_internal/core/log_rotation.cf]: don't

File: cfe_internal/core/log_rotation.cf [hunk 6]
```diff
@@ -0,0 +1,24 @@
+##################################################################
+#
+# cfe_internal_log_rotation
+#  - Rotate logs and clean up old files so the disk doesnt get full
```
Comment (tzz) [inline cfe_internal/core/log_rotation.cf]: doesn't

File: lib/3.7/cfengine_enterprise_hub_ha.cf [hunk 7]
```diff
@@ -51,9 +51,10 @@ bundle common cfengine_enterprise_hub_ha
     enable_cfengine_enterprise_hub_ha::
 
       "input[ha_main]"
-        string => "cfe_internal/ha/ha.cf";
```
Comment (tzz) [inline lib/3.7/cfengine_enterprise_hub_ha.cf]: ha ha

File: promises.cf [hunk 8]
```diff
@@ -131,6 +126,59 @@ bundle common inventory
...
+        string => "cfe_internal/enterprise/CFE_hub_specific.cf",
+        comment => "Policy relating to CFEngine Enterprise Hub, for example
+                    software updates, webserver configuration, and alerts";
+      
+      "input[enterprise_knowledge]"
+        string => "cfe_internal/enterprise/CFE_knowledge.cf",
+        comment => "Settings mostly releated to CFEngine Enteprise Mission Portal";
+
+    any::
+      "inputs" slist => getvalues("input");
```
Comment (tzz) [inline promises.cf]: I like this, it's clever!

File: cfe_internal/CFE_cfengine.cf (hunk 1 above)
Comment (nickanderson) [inline cfe_internal/CFE_cfengine.cf]: TY

File: cfe_internal/CFE_cfengine.cf (hunk 2 above)
Comment (nickanderson) [inline cfe_internal/CFE_cfengine.cf]: TY

File: cfe_internal/core/limit_robot_agents.cf (hunk 3 above)
Comment (nickanderson) [inline cfe_internal/core/limit_robot_agents.cf]: TY

File: lib/3.7/cfengine_enterprise_hub_ha.cf (hunk 7 above)
Comment (nickanderson) [inline lib/3.7/cfengine_enterprise_hub_ha.cf]: https://www.youtube.com/watch?v=rX7wtNOkuHo

File: promises.cf (hunk 8 above)
Comment (nickanderson) [inline promises.cf]: I'm glad you like it, I thought it was clever too, and IMHO a good use of classic arrays.

**Comment by nickanderson** [comment]
@estenberg ping

File: promises.cf [hunk 9]
```diff
@@ -131,6 +116,63 @@ bundle common inventory
...
+        comment => "This policy ensures that various cfengine log files
+                    do not grow without bound and fill up the disk";
+
+    !cfengine_3_5::
+      "input[core_host_info_report]"
+        string => "cfe_internal/core/host_info_report.cf",
+        comment => "This policy produces a text based host info report
+                    and serves as a functional example of using mustache templates";
+
+    enterprise_edition::
```
Comment (estenberg) [inline promises.cf]: As discussed, let's tailor this to the role and not the license.

For example, enterprise_hub_specific and enterprise_knowledge should probably be classified with enterprise.policy_server. There is probably more examples.

File: promises.cf [hunk 10]
```diff
@@ -131,6 +116,63 @@ bundle common inventory
...
+
+      "input[enterprise_main]"
+        string => "cfe_internal/enterprise/main.cf",
+        comment => "This policy activates other enterprise specific policies";
+
+      "input[enterprise_main]"
```
Comment (estenberg) [inline promises.cf]: Duplicate?

**Comment by estenberg** [comment]
Overall looks good, will probably have more tiny suggestions after it's merged and we play with it.

**Comment by kacfengine** [comment]
trigger build

**Comment by kacfengine** [comment]
No detailed review from me. I think with tzz and estenberg's reviews it should be well covered already. I'm just making sure the build works. :-)

**Comment by nickanderson** [comment]
trigger build

**Comment by kacfengine** [comment]
This failed in the autorun.cf acceptance test. This seems to be the relevant section:

```
error: Promised to edit "/home/jenkins/workspace/testing-enterprise-pr_masterfiles/label/PACKAGES_x86_64_linux_redhat_4/masterfiles/tests/acceptance/workdir/__promises_autorun_cf/tmp/TEST.cfengine/var/cfengine/masterfiles/lib/3.7/def.cf", but file does not exist
error: Method "test" failed in some repairs
R: /home/jenkins/workspace/testing-enterprise-pr_masterfiles/label/PACKAGES_x86_64_linux_redhat_4/masterfiles/tests/acceptance/./promises/autorun.cf FAIL
```

Note that this test was fixed recently due to the def.cf reorganization, so a similar fix may be needed again.

**Comment by kacfengine** [comment]
trigger build

**Comment by kacfengine** [comment]
trigger build

**Comment by kacfengine** [comment]
These tests are failing:

* ``masterfiles/tests/acceptance/promises/autorun.cf``
* ``masterfiles/tests/acceptance/promises/autorun-D.cf``
* ``masterfiles/tests/acceptance/promises/promises.cf``

with messages of this kind:

<pre>
error: Finished command related to promiser ""/home/jenkins/workspace/testing-enterprise-pr_masterfiles/label/PACKAGES_x86_64_linux_redhat_4/masterfiles/tests/acceptance/workdir/__promises_promises_cf/bin/cf-promises" -c --eval-functions -f ../../promises.cf" -- an error occurred, returned 1
notice: Q: "...-promises" -c -": ../../cfe_internal/enterprise/main.cf:13:0: error: Undefined bundle cfe_internal_update_folders with type usebundle
Q: "...-promises" -c -": ../../cfe_internal/enterprise/main.cf:20:0: error: Undefined bundle cfe_internal_purge_scheduled_reports_older_than_days with type usebundle
Q: "...-promises" -c -": ../../cfe_internal/enterprise/main.cf:25:0: error: Undefined bundle cfe_internal_webserver with type usebundle
Q: "...-promises" -c -": ../../cfe_internal/enterprise/main.cf:29:0: error: Undefined bundle cfe_internal_setup_knowledge with type usebundle
Q: "...-promises" -c -": ../../cfe_internal/enterprise/main.cf:33:0: error: Undefined bundle cfe_internal_apache_sudoer with type usebundle
Q: "...-promises" -c -": ../../cfe_internal/enterprise/main.cf:37:0: error: Undefined bundle cfe_internal_httpd_related with type usebundle
Q: "...-promises" -c -": ../../cfe_internal/enterprise/main.cf:41:0: error: Undefined bundle cfe_internal_php_runalerts with type usebundle
Q: "...-promises" -c -": ../../cfe_internal/enterprise/main.cf:52:0: error: Undefined bundle cfe_internal_truncate_events with type usebundle
Q: "...-promises" -c -": ../../cfe_internal/enterprise/main.cf:58:0: error: Undefined bundle cfe_internal_postgresql_maintenance with type usebundle
error: Method "test" failed in some repairs
R: /home/jenkins/workspace/testing-enterprise-pr_masterfiles/label/PACKAGES_x86_64_linux_redhat_4/masterfiles/tests/acceptance/./promises/promises.cf FAIL
error: Method "test_run" failed in some repairs
</pre>

**Comment by nickanderson** [comment]
trigger build

**Pull request merged**